src/main/kotlin/tay/example/manage_stock_of_book/service/BookService.kt

```diff
@@ -0,0 +1,161 @@
+                }
+
+                if (limit != 0.toInt()){
+                        books = bookRepository.findAll(PageRequest.of(0,limit))
+                }
+
+                if (page != 0.toInt() && limit != 0.toInt()){
+                        books = bookRepository.findAll(PageRequest.of(page,limit))
+                }
+
+                if (value != "null" && page != 0.toInt()){
+                        books = bookRepository.findListBook(value,PageRequest.of(page,5))
+                }
+
+                if (value != "null" && limit != 0.toInt()){
+                        books = bookRepository.findListBook(value,PageRequest.of(0,limit))
+                }
+
+                if (categoryId != 0.toLong() && page != 0.toInt()){
+                        category = categoryRepository.findById(categoryId!!).orElseThrow {
+                                IDNotFoundException("categoryID")
+                        }
+                        books = bookRepository.findListBook(category!!,PageRequest.of(page,5))
+                }
+
+                if (categoryId != 0.toLong() && limit != 0.toInt()){
+                        category = categoryRepository.findById(categoryId!!).orElseThrow {
+                                IDNotFoundException("categoryID")
+                        }
+                        books = bookRepository.findListBook(category!!,PageRequest.of(0,limit))
+                }
+
+                return ResponseEntity.ok(books!!)
+        }
+
+        fun createBook(createBookDTO: CreateBookDTO): ResponseEntity<Any>{
+                val category = categoryRepository.findById(createBookDTO.category).orElseThrow{
+                        throw IDNotFoundException("categoryID")
+                }
+
+                val book = Book.fromDTO(createBookDTO,category)
+                return ResponseEntity.ok(bookRepository.save(book).toDTO())
+        }
+
+        fun getDetailBook(id: Long): ResponseEntity<Any>{
+                try {
+                        val book = bookRepository.findById(id).get().toDTO()
+                        return ResponseEntity.ok(book)
+                }catch (e: RuntimeException){
+                        throw IDNotFoundException("bookID")
+                }
+
+        }
+
+        fun updateBook(updateBookDTO: UpdateBookDTO, id: Long): ResponseEntity<Any>{
+                val book = bookRepository.findById(id).orElseThrow {
+                        throw IDNotFoundException("bookID")
+                }
+
+//                val category = if (updateBookDTO.category == null)
+//                        categoryRepository.findById(book.category.id).orElseThrow{
+//                                throw IDNotFoundException("categoryID")
+//                        }
+//                else categoryRepository.findById(updateBookDTO.category!!).orElseThrow {
+//                        throw IDNotFoundException("categoryID")
+//                }
+
+                val category = updateBookDTO.let {
```

Comment:
When you use function .let no need to filter if null or not. Just call method find by id and throw ID not Found